util/clustergen/occamy.py

```diff
@@ -20,12 +20,24 @@ def __init__(self, cfg):
         pma_cfg = PMACfg()
         addr_width = cfg["cluster"]['addr_width']
         # Make the entire HBM, but not HBI cacheable
-        pma_cfg.add_region(PMA.CACHED, 0x0000_8000_0000, 0xffff_8000_0000)
-        pma_cfg.add_region(PMA.CACHED, 0x0010_0000_0000, 0xfff0_0000_0000)
+        pma_cfg.add_region_length(PMA.CACHED,
+                                  cfg["hbm"]["address_0"],
+                                  cfg["hbm"]["nr_channels_address_0"] * cfg["hbm"]["channel_size"],
+                                  addr_width)
+        pma_cfg.add_region_length(PMA.CACHED,
+                                  cfg["hbm"]["address_1"],
+                                  cfg["hbm"]["nr_channels_total"] * cfg["hbm"]["channel_size"],
+                                  addr_width)
```

Comment:
changes caused RTL change: see other [comment](https://github.com/pulp-platform/snitch/pull/320/files?authenticity_token=5LqIqefCik%2BESGIAcu366MJA0sTfpx1qluazva1%2FbsiU1KJhz43fa%2Fi1KlYIQ67gr9gTmaYrVg7%2F%2BXwDOs0r7g%3D%3D&file-filters%5B%5D=.py&file-filters%5B%5D=.sv&file-filters%5B%5D=.tex&file-filters%5B%5D=dotfile#r769805452)

Reply:
As mentioned above, to be seen if this affects/degrades implementation quality (hopefully not, since I prefer it on prinicple).